### PR TITLE
Add verify_ssl to OpenStack credential type

### PR DESCRIPTION
**Pull request by saito-hideki**
##### SUMMARY

This PR adds the "Verify SSL" checkbox to the OpenStack authentication information type.
Users can turn it on or off to access Keystone API.

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - UI

##### AWX VERSION

```
- devel
```

##### ADDITIONAL INFORMATION

This PR is related on the issue #941 (RFE: OpenStack Credential Expantion)

### CODE REVIEW

File: awx/main/models/credential/__init__.py [hunk 1]
```diff
@@ -935,6 +935,11 @@ def openstack(cls):
                                            'It is only needed for Keystone v3 authentication '
                                            'URLs. Refer to Ansible Tower documentation for '
                                            'common scenarios.')
+            }, {
+                'id': 'verify_ssl',
+                'label': ugettext_noop('Verify SSL'),
+                'type': 'boolean',
+                'secret': False
```
Comment (ryanpetrello) [inline awx/main/models/credential/__init__.py]: This is redundant and you can remove it; booleans are never secret.

**Comment by ryanpetrello** [comment]
cc @AlanCoding

**Comment by ryanpetrello** [comment]
@saito-hideki this looks great - thanks for contributing this!

Have you actually run this and tested it against an OpenStack deployment?

**Review by ryanpetrello — APPROVED** [review]

File: awx/main/tasks.py [hunk 2]
```diff
@@ -1143,6 +1143,7 @@ def build_private_data(self, job, **kwargs):
                     'clouds': {
                         'devstack': {
                             'auth': openstack_auth,
+                            'verify': credential.get_input('verify_ssl', default=False),
```
Comment (AlanCoding) [inline awx/main/tasks.py]: was False the default before?

**Comment by AlanCoding** [comment]
@jladdjr we need to rebase the `inventory_plugins` branch right after this lands.

**Comment by AlanCoding** [comment]
There are 2 comments here that I would like to see addressed, after that I'm okay with merging.

File: awx/main/models/credential/__init__.py (hunk 1 above)
Comment (saito-hideki) [reply awx/main/models/credential/__init__.py]: Thank you for the review! I will remove `'secret': False` from this section :)

File: awx/main/tasks.py (hunk 2 above)
Comment (saito-hideki) [reply awx/main/tasks.py]: Thank you for the review!
I confirmed it. if we do not specify the `verify` parameter, it behaves as` TRUE`. 
I'll trying to modify it to set `TRUE` by default.

**Comment by saito-hideki** [comment]
@ryanpetrello @AlanCoding I have finished to modify this PR based on your review.

- Remove `secret: false` from `openstack().inputs`
- Use `no_verify_ssl` instead of `verify_ssl`
- Rename django migration sequence from 0057 to 0059 to avoid merge conflict

Those fixes worked without errors against my devstack environment via HTTPS.

**Comment by saito-hideki** [comment]
Now I'm working to fix CI failure

**Comment by saito-hideki** [comment]
@ryanpetrello @AlanCoding I have fixed unit-test and finished testing without errors, therefore if it is possible, can you review this PR ? :)

**Comment by AlanCoding** [comment]
It looks like the git history of this got messed up. What git commands did you use to do a rebase? The general practice is that we want things to be rebased to the tip of `devel` (within reason, being out-of-date isn't a problem if there are no conflicts). Something here just went haywire and we're going to need to do some debugging to fix it.

**Comment by saito-hideki** [comment]
@AlanCoding Sorry for that. I'm not familiar with git operation. So I think I made some mistake at that time.
But I have a complete patch of this PR. I'm considering applying this to the latest upstream `devel`, then re-create a new PR. Can I do this ?
It is simple way for me.

**Comment by AlanCoding** [comment]
If you were to manually re-apply all your changes, then I have no doubt that will work. We could approve that in a separate PR and merge it, yes.

It will probably still be easier to resolve the problem using git, but back up your work first `git checkout -b pr/openstack_cred_verify_ssl_backup`, in case you mess up. Try `git pull --rebase origin devel`, try squashing your work `git rebase -i <hash of commit before your work started>`. You can get help if you need it.

**Comment by saito-hideki** [comment]
@ryanpetrello @AlanCoding  Thank you for your cooperation, unfortunately, I could not fix my local repository. So I will close this PR and create a separate PR.

**Pull request closed without merging**